### PR TITLE
Clarify development procedures

### PR DESCRIPTION
**Pull request by dchiquito**
I forgot to clarify the development process in the original PR, here are those changes.

### CODE REVIEW

File: doc/design/staging-1.md [hunk 1]
```diff
@@ -30,12 +30,22 @@ To test code changes to the web GUI, branch previews will also be pointed at the
 
 The production web GUI will track the `master` branch in `dandiarchive` and will automatically redeploy whenever `master` is changed.
 
-* A developer has a feature they want to add that impacts both the backend and the frontend.
+A developer has a feature they want to add that impacts both the backend and the frontend.
 * They checkout a new branch in `dandi-api` and `dandiarchive`, do the work, and submit related PRs in both repos.
```
Comment (yarikoptic) [inline doc/design/staging-1.md:34]: both against `master` branch? (it is better to state branches at this step since PR needs to know the base branch to start with, so better to state here and remove branches mentioning below for "PR is merged into ...")

File: doc/design/staging-1.md [hunk 2]
```diff
@@ -30,12 +30,22 @@ To test code changes to the web GUI, branch previews will also be pointed at the
...
-* Once approved, the `dandi-api` PR is merged into `master` and the `dandiarchive` PR is merged into `staging`.
-* The changes are tested in staging.
-* Once approved, a new release of `dandi-api` is tagged, pushing the changes to production.
-* Once approved, the developer manually merges `dandiarchive/staging` into `dandiarchive/master`.
+* Once approved, the `dandi-api` PR is merged into `master`.
+* The `dandiarchive` changes are tested in the branch preview, which points at the staging environment.
+* Once `dandiarchive` is approved, the developer merges the PR, and tags a new release of `dandi-api`, which is then automatically deployed to production.
```
Comment (yarikoptic) [inline doc/design/staging-1.md:37]:
```suggestion
* Once `dandiarchive` PR is approved, the developer merges the PR, and tags a new release of `dandi-api`, which is then automatically deployed to production.
```

File: doc/design/staging-1.md [hunk 3]
```diff
@@ -30,12 +30,22 @@ To test code changes to the web GUI, branch previews will also be pointed at the
 
 The production web GUI will track the `master` branch in `dandiarchive` and will automatically redeploy whenever `master` is changed.
 
-* A developer has a feature they want to add that impacts both the backend and the frontend.
+A developer has a feature they want to add that impacts both the backend and the frontend.
```
Comment (yarikoptic) [inline doc/design/staging-1.md:33]: in above text, not part of the PR, so can't comment there:
```
The staging web GUI will track the `master` branch in `dandiarchive` and will automatically redeploy whenever `master` is changed.
...
The production web GUI will track the `master` branch in `dandiarchive` and will automatically redeploy whenever `master` is changed.
```
doesn't sound right since they look identical to me and thus in effect we do not have "staging" (beyond state of PR) for "web GUI", is that the intention now (sorry if I didn't chime in time on that discussion). Or is that the current plan?

Comment (yarikoptic) [reply doc/design/staging-1.md:33]: if that it the plan, might be better to have a single sentence like
```
Both staging and production web GUI will track the `master` branch in `dandiarchive` and will automatically redeploy whenever `master` is changed.
```

Comment (dchiquito) [reply doc/design/staging-1.md:33]: The next sentence for the staging GUI description clarifies:
```
The configuration variables for the staging Netlify deployment will point it at the staging server rather than the production server.
```
So there two permanent GUI servers that are running the exact same frontend code, but are pointing at different backends.

If you want to test frontend changes against staging, you would use a branch preview, which is also pointed at the staging API server.

File: doc/design/staging-1.md (hunk 1 above)
Comment (dchiquito) [reply doc/design/staging-1.md:34]: I have removed any reference in this file to any other branch except `master`, so I think it would be implicit to anyone reading the document. If anyone is confused about where to base the PR when following these instructions, I don't think it's unreasonable for them to read one step ahead for the answer.

**Review by waxlamp — APPROVED** [review]
